### PR TITLE
Fixed issue #17500: When mass-deleting, list survey names and ids to be deleted

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/controllers/admin/surveyadmin.php [hunk 1]
```diff
@@ -75,6 +75,34 @@ public function deleteMultiple()
         );
     }
 
+    /**
+     * render selected items for massive action
+     * @return void
+     */
+    public function renderItemsSelected()
+    {
+        $aSurveys = json_decode(Yii::app()->request->getPost('$oCheckedItems'));
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/controllers/admin/surveyadmin.php:84]: `json_decode(Yii::app()->request->getPost('$oCheckedItems'), true);`

Must return an array.

**Review by Shnoulle — CHANGES_REQUESTED** [review]
1. json decode as array,
2. Must check if it's a valid array
3. Must check if it's valid surveyid

File: application/controllers/admin/surveyadmin.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/controllers/admin/surveyadmin.php:84]: Although I hardly believe there could be an error arising from that line, I understand your point.
Still, if we return an error, the ajax is not prepared to handle it (happens with most of the ajax).

Let me see what we can think ok...

File: application/controllers/admin/surveyadmin.php [hunk 2]
```diff
@@ -76,30 +76,33 @@ public function deleteMultiple()
...
-        foreach ($aSurveys as $iSurveyID) {
-            if (Permission::model()->hasSurveyPermission($iSurveyID, 'survey', 'delete')) {
-                $oSurvey                        = Survey::model()->findByPk($iSurveyID);
-                $aResults[$iSurveyID]['title']  = $oSurvey->correct_relation_defaultlanguage->surveyls_title;
-                $aResults[$iSurveyID]['result'] = 'selected';
+        $surveyIds = json_decode(Yii::app()->request->getPost('$oCheckedItems'), true);
+        if(!is_array($surveyIds)) {
+            throw new CHttpException(400, gT('Invalid list of checked items'));
+        }
+        $results = [];
```
Comment (Shnoulle) [inline application/controllers/admin/surveyadmin.php:88]: `$results `

File: application/controllers/admin/surveyadmin.php [hunk 3]
```diff
@@ -76,30 +76,33 @@ public function deleteMultiple()
...
+
+        $tableLabels = [gT('Survey ID'), gT('Survey Title'), gT('Status')];
+        foreach ($surveyIds as $surveyId) {
+            if (Permission::model()->hasSurveyPermission($surveyId, 'survey', 'delete')) {
+                $survey                        = Survey::model()->findByPk($surveyId);
+                $aResults[$surveyId]['title']  = $survey->correct_relation_defaultlanguage->surveyls_title;
```
Comment (Shnoulle) [inline application/controllers/admin/surveyadmin.php:94]: `$aResults`

File: application/controllers/admin/surveyadmin.php [hunk 4]
```diff
@@ -75,6 +75,37 @@ public function deleteMultiple()
...
+                $aResults[$surveyId]['result'] = 'selected';
+            }
+        }
+
+        Yii::app()->getController()->renderPartial(
+            'ext.admin.survey.ListSurveysWidget.views.massive_actions._selected_survey',
+            [
+                'aResults'     => $results,
```
Comment (Shnoulle) [inline application/controllers/admin/surveyadmin.php:102]: `$results `

**Review by Shnoulle — CHANGES_REQUESTED** [review]
$results != $aResults ,

Seems you set and return `$results` but update `$aResults`

Can you check please ?

File: application/controllers/admin/surveyadmin.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/controllers/admin/surveyadmin.php]: Using POST value : always have potential issue by user value. All user value must be checked :)

**Review by Shnoulle — CHANGES_REQUESTED** [review]
Set `$results `
Update `$aResults`
use ` $results`

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**